shared/src/persistence/elasticsearch/getReconciliationReport.js

```diff
@@ -0,0 +1,69 @@
+const {
+  SERVED_PARTIES_CODES,
+} = require('../../business/entities/EntityConstants');
+const { omit } = require('lodash');
+const { search } = require('./searchClient');
+
+/**
+ * getReconciliationReport
+ * all items served on the IRS (indicated by servedParty of R or B) on a specific day (12:00am-11:59:59pm ET)
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @returns {object} the docket entries
+ */
+exports.getReconciliationReport = async ({
+  applicationContext,
+  reconciliationDateEnd,
+  reconciliationDateStart,
+}) => {
+  const query = {
+    bool: {
+      must: [
+        {
+          terms: {
+            'servedPartiesCode.S': [
+              SERVED_PARTIES_CODES.RESPONDENT,
+              SERVED_PARTIES_CODES.BOTH,
+            ],
+          },
+        },
+        {
+          range: {
+            'servedAt.S': {
+              format: 'strict_date_time', // ISO-8601 time stamp
+              gte: reconciliationDateStart,
+              lte: reconciliationDateEnd,
+            },
+          },
+        },
+      ],
+    },
+  };
+
+  const { results } = await search({
+    applicationContext,
+    searchParameters: {
+      body: {
+        _source: [
+          'pk',
+          'docketNumber',
+          'documentTitle',
+          'docketEntryId',
+          'eventCode',
+          'filedBy',
+          'filingDate',
+          'caseCaption',
+          'servedAt',
+          'servedPartiesCode',
+        ],
+        query,
+        size: 5000,
```

Comment:
Something to keep an eye on. What happens if we serve more than 5000 documents? Based on my observations in Mig, this number is around 1,000. 

Nothing that requires changing because I think this should suffice, but we may need to consider what happens if the total results is > 5000. Yes, we can increase to 10k or use ES's scroll feature. And we may need to consider pagination in the report itself.